### PR TITLE
Improve params parser

### PR DESCRIPTION
**Pull request by rodzyn**
While playing with `ParamsParser` I found some stuff that might be cool to change.

### CODE REVIEW

**Comment by seuros** [comment]
could you squash your commits ?

**Comment by rodzyn** [comment]
@seuros done

File: actionpack/lib/action_dispatch/middleware/params_parser.rb [hunk 1]
```diff
@@ -28,16 +39,16 @@ def initialize(app, parsers = {})
     def call(env)
       request = Request.new(env)
 
-      request.request_parameters = parse_formatted_parameters(request, @parsers)
+      request.request_parameters = parse_formatted_parameters(request)
```
Comment (rafaelfranca) [inline actionpack/lib/action_dispatch/middleware/params_parser.rb]: Could you undo this change? We are passing the instance variable by design. See https://github.com/rails/rails/commit/f3bae24c81e7198d9bbedaf79056d9961e59113c

File: actionpack/lib/action_dispatch/middleware/params_parser.rb [hunk 2]
```diff
@@ -13,6 +19,7 @@ def initialize(message, original_exception)
       end
     end
 
+    #The default parsers list includes just JSON parser
```
Comment (rafaelfranca) [inline actionpack/lib/action_dispatch/middleware/params_parser.rb]: space after `#`

File: actionpack/lib/action_dispatch/middleware/params_parser.rb [hunk 3]
```diff
@@ -21,6 +28,10 @@ def initialize(message, original_exception)
       }
     }
 
+    # Create a new +RemoteIp+ middleware instance.
```
Comment (matthewd) [inline actionpack/lib/action_dispatch/middleware/params_parser.rb]: This is not true

File: actionpack/lib/action_dispatch/middleware/params_parser.rb [hunk 4]
```diff
@@ -1,9 +1,15 @@
...
 require 'action_dispatch/http/request'
-require 'active_support/core_ext/hash/indifferent_access'
 
 module ActionDispatch
+  # ActionDispatch::ParamsParser works for all the requests having any Content-Length
+  # (like POST). It takes raw data from the request and puts it through the parser
+  # that is picked based on Content-Type header.
+  #
+  # In case of any error while parsing data ParamsParser::ParseError is raised.
+
```
Comment (rafaelfranca) [inline actionpack/lib/action_dispatch/middleware/params_parser.rb]: :scissors: this blank line

File: actionpack/lib/action_dispatch/middleware/params_parser.rb [hunk 5]
```diff
@@ -1,9 +1,15 @@
-require 'active_support/core_ext/hash/conversions'
 require 'action_dispatch/http/request'
-require 'active_support/core_ext/hash/indifferent_access'
```
Comment (matthewd) [inline actionpack/lib/action_dispatch/middleware/params_parser.rb]: Are these no longer used?

I can't see any explanation for why you're making these changes. (Maybe because @seuros asked you to squash it away, before anyone had a chance to look?)

File: actionpack/lib/action_dispatch/middleware/params_parser.rb (hunk 2 above)
Comment (matthewd) [inline actionpack/lib/action_dispatch/middleware/params_parser.rb]: We should remove this; it's documenting something that's both obvious from the code, and not something we have any reason to commit to in our API.

File: actionpack/lib/action_dispatch/middleware/params_parser.rb (hunk 5 above)
Comment (rodzyn) [inline actionpack/lib/action_dispatch/middleware/params_parser.rb]: @matthewd I don't see them used here anywhere

File: actionpack/lib/action_dispatch/middleware/params_parser.rb (hunk 1 above)
Comment (rodzyn) [inline actionpack/lib/action_dispatch/middleware/params_parser.rb]: @rafaelfranca done

File: actionpack/lib/action_dispatch/middleware/params_parser.rb (hunk 2 above)
Comment (rodzyn) [inline actionpack/lib/action_dispatch/middleware/params_parser.rb]: @matthewd removed

File: actionpack/lib/action_dispatch/middleware/params_parser.rb [hunk 6]
```diff
@@ -1,9 +1,14 @@
...
 
 module ActionDispatch
+  # ActionDispatch::ParamsParser works for all the requests having any Content-Length
+  # (like POST). It takes raw data from the request and puts it through the parser
+  # that is picked based on Content-Type header.
+  #
+  # In case of any error while parsing data ParamsParser::ParseError is raised.
   class ParamsParser
+    # Raised when raw data from the request cannot be parsed by the parser
+    # defined for request's content mime type
```
Comment (rafaelfranca) [inline actionpack/lib/action_dispatch/middleware/params_parser.rb]: `.` in the end

File: actionpack/lib/action_dispatch/middleware/params_parser.rb [hunk 7]
```diff
@@ -21,6 +26,10 @@ def initialize(message, original_exception)
       }
     }
 
+    # Create a new +ParamsParser+ middleware instance.
+    #
+    # The +parsers+ argument can take Hash of parsers where key is identifying
+    # content mime type, and value is a lambda that is going to process data
```
Comment (rafaelfranca) [inline actionpack/lib/action_dispatch/middleware/params_parser.rb]: `.` in the end

Comment (rodzyn) [inline actionpack/lib/action_dispatch/middleware/params_parser.rb]: @rafaelfranca thanks for catching, changed

**Comment by rafaelfranca** [comment]
Thank you!
